node/src/store/in_memory_store.rs

```diff
@@ -211,6 +216,26 @@ impl Store for InMemoryStore {
         self.get_by_height(height)
     }
 
+    async fn wait_height(&self, height: u64) -> Result<()> {
+        let mut notifier = pin!(self.check_height_notifier.notified());
+
+        loop {
+            // Make sure no notifications are lost by enable notifier
+            // before the check.
+            notifier.as_mut().enable();
+
```

Comment:
according to docs, the `Notified::enable` is only needed in case we don't want to lost the notifications coming from `notify_one`. As we are using the `notify_waiters` this has no effect

Reply:
true, removed